### ISSUE TITLE
Widgets preview uses a relative URL that breaks when the page is moved

### ISSUE THREAD

**Issue by tomjn** [WordPress/gutenberg#30049]
## Description

The experimental widgets UI has a preview function, **however this preview uses a relative URL**.

Normally this is fine, however, I've been trying to add this to frontenberg, which means the code is no longer running from the `/wp-admin/` URL. As a result all previews are now Nginx 404 error pages on my local environment.

## Expected behaviour

A full URL, e.g.

```
<iframe src="https://one.wordpress.test/wp-admin/themes.php?page=gutenberg-widgets&amp;widget-preview%5BwidgetClass%5D=WP_Widget_Recent_Comments&amp;widget-preview%5Bname%5D=Recent%20Comments&amp;widget-preview%5BidBase%5D=recent-comments&amp;widget-preview%5Bnumber%5D=2&amp;widget-preview%5Binstance%5D%5Btitle%5D=&amp;widget-preview%5Binstance%5D%5Bnumber%5D=5&amp;widget-preview%5B__internalWidgetId%5D=recent-comments-2&amp;widget-preview%5BsidebarId%5D=wp_inactive_widgets" height="138" class="wp-block-legacy-widget__preview"></iframe>
```

## Actual behaviour

`src="themes.php?....` is used, resulting in this:

<img width="746" alt="Screenshot 2021-03-19 at 22 55 00" src="https://user-images.githubusercontent.com/58855/111849713-29430800-8906-11eb-889e-ac6b0303fdff.png">

## WordPress information
- WordPress version: 5.7.0
- Gutenberg version: 10.2.0
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? No ( modified Frontenberg )

## Device information
- Device: MBP 13" Intel
- Operating system: MacOS 11.2
- Browser: Chrome 89

**Comment by noisysocks** [comment]
Thanks for the bug report. Let's use the full URL. We will have to pass the WP Admin URL via block editor settings to the block.

**Comment by anton-vlasenko** [comment]
@tomjn I'm sorry, I can't reproduce this on https://frontenberg.tomjn.com
Could you please provide some steps on how to reproduce this?
FYI: @kevin940726

**Comment by tomjn** [comment]
Frontenberg doesn't advertise widgets functionality because of this issue. The iframes use relative URLs, they should use absolute URLs or at a minimum URLs that are relative to `/`

**Comment by tomjn** [comment]
In particular, it tries to use URLs like this:

`https://frontenberg.tomjn.com/widgets.php?legacy-widget-preview%5BidBase%5D=akismet_widget&legacy-widget-preview%5Binstance%5D%5Bencoded%5D=YToxOntzOjU6InRpdGxlIjtzOjA6IiI7fQ%3D%3D&legacy-widget-preview%5Binstance%5D%5Bhash%5D=3a1e89203a295a9a3374f02b7b41b341`

But obviously there is no `widgets.php` file in the root so it fails. Notice that `widgets.php` is a relative URL, the code _assumes_ the current page and URL to be `/wp-admin`, which is incorrect.

Here is where the assumption is made in the JS:

https://github.com/WordPress/gutenberg/blob/fea614a3ac1f38cfa00589d66553603215f4df2f/packages/widgets/src/blocks/legacy-widget/edit/preview.js#L99

**Comment by tomjn** [comment]
This code may be useful to anybody trying to correct the issue:

https://github.com/WordPress/gutenberg/blob/be48e651c5c94333fabea93be7c5273085b89564/packages/e2e-test-utils/src/visit-admin-page.js#L21

To reproduce the issue, load a legacy widgets UI in any location that is not a file in `wp-admin`. e.g. `/wp-admin/test`, `/`, `/test`, etc

**Comment by tomjn** [comment]
@kevin940726 was this reverted? The issue has reappeared and there is no trace of the fix in the `trunk` branch

**Comment by kevin940726** [comment]
@tomjn Yep the fix is still there on trunk. It appears that it was not ported in 5.8.1 though, but should be available in Gutenberg 11.5, which hasn't released yet.
